pkg/host-disk/host-disk.go

```diff
@@ -88,18 +88,26 @@ func dirBytesAvailable(path string) (uint64, error) {
 	return stat.Bavail * uint64(stat.Bsize), nil
 }
 
-func createSparseRaw(fullPath string, size int64) error {
+func createSparseRaw(fullPath string, size int64) (err error) {
+	var f *os.File
 	offset := size - 1
-	f, err := os.Create(fullPath)
-	defer f.Close()
+	f, err = os.Create(fullPath)
+
+	defer func() {
+		ferr := f.Close()
+		if err == nil {
+			err = ferr
```

Comment:
This can hide some of the errors. This is probably an issue with Go's implementation of `file.Close`, but if the creation fails, you can't call close on it as you will get an `ErrInvalid` back.

One solution is to move this defer to line 105, after we ensure that the file was created.

Reply:
Which errors this hides?  I explicitly check to see if there are other errors and if so we don't care about the close() errors and I don't override the `err` variable

Reply:
Fixed